esp-hal-common/src/spi.rs

```diff
@@ -1665,7 +1667,20 @@ where
         tx: &mut TX,
         rx: &mut RX,
     ) -> Result<&'w [u8], Error> {
-        for chunk in words.chunks_mut(MAX_DMA_SIZE) {
+        let (max, copy) = if !crate::soc::is_valid_ram_address(&words[0] as *const _ as u32) {
```

Comment:
I think this impl is redundant actually, because I think ` transfer_in_place_dma` will never be used with a flash buffer, and if it does, its probably an error anyway.